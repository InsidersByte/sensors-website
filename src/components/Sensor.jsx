/* @flow */

import React from 'react';
import styled from 'styled-components';
import SensorGraph from './SensorGraph';
import { FULL_DATE_FORMAT } from '../constants';
import type { SensorType } from '../types';

const Wrapper = styled.div`
    box-sizing: border-box;
    width: 30%;
    margin-top: 1.5em;
    padding: 1.5em 0.5em;
    background: #61BFAD;
    color: #FFFFFF;
    text-align: center;
    
    @media (max-width: 1279px) {
        width: 45%;
    }
    
    @media (max-width: 736px) {
        width: 90%;
    }
`;

const Title = styled.div`
    text-transform: capitalize;
    font-size: 2.5em;
`;

const Value = styled.div`
    font-size: 5em;
    margin: 0.1em 0;
`;

const Text = styled.div`
    font-size: 1.25em;
`;

const Sensor = ({ sensor: { name, data } }: { sensor: SensorType }) => {
    const [{ time, value }] = data;

    return (
        <Wrapper>
            <Title>{name}</Title>

            <SensorGraph data={data} />

            <Text>Lastest Reading</Text>
            <Value>{value}</Value>
            <Text>{time.utc().format(FULL_DATE_FORMAT)}</Text>
        </Wrapper>
    );
};

export default Sensor;
