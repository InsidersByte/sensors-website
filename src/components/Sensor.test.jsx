import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import Sensor from './Sensor';

describe('Sensor', () => {
    it('should render', () => {
        const wrapper = shallow(
            <Sensor sensor={{
                name: 'sensor',
                data: [
                    { time: moment.unix(1472120033), value: 25.3 },
                    { time: moment.unix(1472119853), value: 25.1 },
                ],
            }} />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
