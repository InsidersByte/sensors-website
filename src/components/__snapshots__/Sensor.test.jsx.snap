exports[`Sensor should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <styled.div>
        sensor
    </styled.div>
    <SensorGraph
        data={
            Array [
                Object {
                  "time": "2016-08-25T10:13:53.000Z",
                  "value": 25.3,
                },
                Object {
                  "time": "2016-08-25T10:10:53.000Z",
                  "value": 25.1,
                },
              ]
        } />
    <styled.div>
        Lastest Reading
    </styled.div>
    <styled.div>
        25.3
    </styled.div>
    <styled.div>
        25th August 16, 11:13:53
    </styled.div>
</styled.div>,
  "nodes": Array [
    <styled.div>
      <styled.div>
            sensor
      </styled.div>
      <SensorGraph
            data={
                  Array [
                        Object {
                          "time": "2016-08-25T10:13:53.000Z",
                          "value": 25.3,
                        },
                        Object {
                          "time": "2016-08-25T10:10:53.000Z",
                          "value": 25.1,
                        },
                      ]
            } />
      <styled.div>
            Lastest Reading
      </styled.div>
      <styled.div>
            25.3
      </styled.div>
      <styled.div>
            25th August 16, 11:13:53
      </styled.div>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Sensor
        sensor={
                Object {
                        "data": Array [
                          Object {
                            "time": "2016-08-25T10:13:53.000Z",
                            "value": 25.3,
                          },
                          Object {
                            "time": "2016-08-25T10:10:53.000Z",
                            "value": 25.1,
                          },
                        ],
                        "name": "sensor",
                      }
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "sensor": Object {
            "data": Array [
              Object {
                "time": "2016-08-25T10:13:53.000Z",
                "value": 25.3,
              },
              Object {
                "time": "2016-08-25T10:10:53.000Z",
                "value": 25.1,
              },
            ],
            "name": "sensor",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <styled.div>
                    sensor
          </styled.div>
          <SensorGraph
                    data={
                              Array [
                                        Object {
                                          "time": "2016-08-25T10:13:53.000Z",
                                          "value": 25.3,
                                        },
                                        Object {
                                          "time": "2016-08-25T10:10:53.000Z",
                                          "value": 25.1,
                                        },
                                      ]
                    } />
          <styled.div>
                    Lastest Reading
          </styled.div>
          <styled.div>
                    25.3
          </styled.div>
          <styled.div>
                    25th August 16, 11:13:53
          </styled.div>
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <styled.div>
                    sensor
          </styled.div>
          <SensorGraph
                    data={
                              Array [
                                        Object {
                                          "time": "2016-08-25T10:13:53.000Z",
                                          "value": 25.3,
                                        },
                                        Object {
                                          "time": "2016-08-25T10:10:53.000Z",
                                          "value": 25.1,
                                        },
                                      ]
                    } />
          <styled.div>
                    Lastest Reading
          </styled.div>
          <styled.div>
                    25.3
          </styled.div>
          <styled.div>
                    25th August 16, 11:13:53
          </styled.div>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Sensor
    sensor={
        Object {
            "data": Array [
              Object {
                "time": "2016-08-25T10:13:53.000Z",
                "value": 25.3,
              },
              Object {
                "time": "2016-08-25T10:10:53.000Z",
                "value": 25.1,
              },
            ],
            "name": "sensor",
          }
    } />,
}
`;
